packages/shared/src/utils/generate-joi.ts

```diff
@@ -244,13 +266,17 @@ export function generateJoi(filter: FieldFilter, options?: JoiOptions): AnySchem
 			} else {
 				const wrapped =
 					typeof compareValue === 'string' ? compareValue.startsWith('/') && compareValue.endsWith('/') : false;
-				schema[key] = getStringSchema().regex(new RegExp(wrapped ? compareValue.slice(1, -1) : compareValue));
+				schema[key] = getStringSchema().regex(new RegExp(wrapped ? (compareValue as any).slice(1, -1) : compareValue));
 			}
 		}
 	}
 
 	schema[key] = schema[key] ?? Joi.any();
 
+	if (options.allowUnknown) {
+		return Joi.object(schema).unknown();
```

Comment:
The return of this function on 284 is already adding `.unknown()`, making this option useless (the result is the same whether or not `allowUnknown` is true or false. We should make sure where that allowUnknown was used before, and ensure that the new behavior is consistent and not causing any new errors.

Reply:
`allowUnknown` is currently always `true` for both the app & api. 
It was used in 
https://github.com/directus/directus/blob/d8ae3b9a8d3708e0c198d2d05555cfc116b72cf4/app/src/modules/collections/routes/detail/detail.vue#L414-L416

The option remains so that we could reuse to check for exact matches without unknown fields.

Reply:
@jaycammarano I've defaulted `allowUnknown` to be true as I haven't seen it being set to false anywhere. Please assist to verify if that's correct. 👀

Reply:
Removed unused `allowUnknown` option.